servicetalk-opentelemetry-http/src/main/java/io/servicetalk/opentelemetry/http/OpenTelemetryOptions.java

```diff
@@ -109,58 +139,82 @@ public int hashCode() {
 
     @Override
     public String toString() {
-        return getClass().getSimpleName() +
-                "{capturedRequestHeaders=" + capturedRequestHeaders +
+        return "OpenTelemetryOptions{" +
+                "capturedRequestHeaders=" + capturedRequestHeaders +
                 ", capturedResponseHeaders=" + capturedResponseHeaders +
                 ", enableMetrics=" + enableMetrics +
+                ", openTelemetry=" + openTelemetry +
+                ", componentName='" + componentName + '\'' +
                 '}';
     }
 
-    /**
-     * A builder for {@link OpenTelemetryOptions}.
-     */
+    /** A builder for {@link OpenTelemetryOptions}. */
     public static final class Builder {
         private List<String> capturedRequestHeaders = emptyList();
         private List<String> capturedResponseHeaders = emptyList();
         private boolean enableMetrics;
+        private OpenTelemetry openTelemetry = GlobalOpenTelemetry.get();
+
+        // Client-specific options (ignored by server filters)
+        private String componentName = "";
+
+        /**
+         * Create a copy of an existing builder.
+         * @param openTelemetryOptions the {@link Builder} to copy.
+         */
+        public Builder(OpenTelemetryOptions openTelemetryOptions) {
```

Comment:
If we go this route, does this ctor have to be public? Consider starting with a pkg-private visibility

Reply:
It doesn't have to be, it just seemed nice to have.

Reply:
The point is now moot.